[MIG] 14.0 purchase_sale_inter_company

PierrickBrun opened the pull request:
Depends on https://github.com/OCA/multi-company/pull/257

chafique-delli committed:
[ADD] purchase_sale_inter_company module

chafique-delli committed:
add warehouse_id in the values for create sale order

chafique-delli committed:
refactor code after add dependency on account_invoice_inter_company

chafique-delli committed:
check product is intercompany

chafique-delli committed:
extend invoice_method field and add action_cancel method for purchase and auto_purchase_line_id field in sale_order_line

chafique-delli committed:
rename inter_company_po_to_so_view.xml to res_company_view.xml and add purchase_order_view.xml

chafique-delli committed:
use product_id_change_with_wh method for return sale line data

chafique-delli committed:
fix README file

chafique-delli committed:
remove inter company user

chafique-delli committed:
remove unnecessary dependencies in openerp.py and update of the unit price in purchase line when validating the sale order

chafique-delli committed:
add demo data and tests

chafique-delli committed:
fix partner_shipping_id in _prepare_sale_order_data method

chafique-delli committed:
add force_company in context when create inter company sale oder

sebastienbeau committed:
[FIX] fix setting the force company pass the cest_company as an id in order to load it with the correct context

chafique-delli committed:
replace sale_line_vals by sale_line_data

chafique-delli committed:
fix tests

chafique-delli committed:
fix following @elicoidal's comments and add french translation

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

chafique-delli committed:
fix domain for search users in _check_intercompany_product method

chafique-delli committed:
fix multi-company access rights issue

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[MIG] purchase_sale_inter_company: Migration to 11.0 (from 8.0)

carlosdauden committed:
[IMP] purchase_sale_inter_company: Recover intercompany user

pedrobaeza committed:
[FIX] purchase_sale_inter_company: Transfer dropship pickings correctly

carlosdauden committed:
[IMP] purchase_sale_inter_company: Better relation between docs. Translate.

yajo committed:
[FIX] purchase_sale_inter_company: No buggy demo data

Make tests compatible with https://github.com/OCA/multi-company/pull/110, for the same reason as the specified in that link.

chienandalu committed:
[FIX] tests and other issues

mourad-ehm committed:
[MIG] purchase_sale_inter_company: Migration to 12.0

Commit:
Update purchase_sale_inter_company/README.rst OCA Review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA Review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

mourad-ehm committed:
[FIX] OCA Review

mourad-ehm committed:
[FIX] ref uom.product_uom_unit

mourad-ehm committed:
[FIX] import Warning

mourad-ehm committed:
[FIX] import Warning (To squash)

mourad-ehm committed:
[FIX] demo test data

mourad-ehm committed:
[FIX] add demo product

mourad-ehm committed:
[FIX] product_qty

mourad-ehm committed:
[FIX] unit test

mourad-ehm committed:
[FIX] pylint

mourad-ehm committed:
[FIX] set readonly = false for  all res_config_settings fields

bealdav committed:
IMP purch_sale_inter_cpny: layout settings

oca-travis committed:
[UPD] Update purchase_sale_inter_company.pot

eLBati committed:
Added translation using Weblate (Italian)

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (25 of 25 strings)

Translation: multi-company-12.0/multi-company-12.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-purchase_sale_inter_company/it/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-12.0/multi-company-12.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-purchase_sale_inter_company/

AdriaGForgeFlow committed:
[IMP] purchase_sale_inter_company: black, isort

AdriaGForgeFlow committed:
[13.0][MIG] purchase_sale_inter_company

rename intercompany_user_id to intercompany_sale_user_id for compatibility with account_invoice_inter_company

[IMP] res_config_settings view

oca-travis committed:
[UPD] Update purchase_sale_inter_company.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/

dong-z committed:
Added translation using Weblate (Chinese (Simplified))

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 14.8% (4 of 27 strings)

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/zh_CN/

AaronHForgeFlow committed:
[FIX]purchase_sale_inter_company tests
* product multi company has company because of the default method

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/

AaronHForgeFlow committed:
[IMP] purchase_sale_inter_company
* move demo data to SetupClass

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.1

chienandalu committed:
[FIX] purchase_sale_inter_company: commitment_date

The field was renamed from `requested_date` to `commitment_date`

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.2

Commit:
[MIG] purchase_sale_inter_company: Migration to 14.0

Joaco1980 reviewed (approved):
It works perfectly v14.0

enriquemartin committed:
[FIX] purchase_sale_inter_company: propagate correct uom to sale.order.line

Commit:
[13.0] formatting

enriquemartin commented on purchase_sale_inter_company/models/purchase_order.py:
```diff
@@ -0,0 +1,182 @@
+        )
+        for onchange_method in new_line._onchange_methods["product_id"]:
+            onchange_method(new_line)
+        return new_line._convert_to_write(new_line._cache)
```
`product_uom` still needs rewritten because of this: https://github.com/OCA/multi-company/pull/268

`        new_line.update({
            "product_uom": purchase_line.product_uom.id,
        })`

enriquemartin commented on purchase_sale_inter_company/models/sale_order.py:
```diff
@@ -0,0 +1,35 @@
+        for order in self.filtered("auto_purchase_order_id"):
+            for line in order.order_line.sudo():
+                if line.auto_purchase_line_id:
+                    line.auto_purchase_line_id.price_unit = line.price_unit
```
I don't understand this; the user does not expect a change in the price entered in the purchase line, regardless of whether the sale price is different or not. Am I missing something?
If we want to mantain the sale price maybe we should at least log some kind of message to inform the user. There is a simililar behaviour in [account_invoice_inter_company/models/account_move.py#L125](https://github.com/OCA/multi-company/blob/16031b59e8021c32f708a79a24160aa1466b97f0/account_invoice_inter_company/models/account_move.py#L125)

StefanRijnhart replied:
I agree with your comment, and I have a customization in place that communicates price changes between both orders and then refuses to confirm in case of a price difference, but this code has been there from the first incarnation of this module in the 8.0 branch so this is something that is best addressed in a follow-up PR and not in this migration PR.

StefanRijnhart replied:
Here is a PR for that: https://github.com/OCA/multi-company/pull/307

enriquemartin reviewed (changes requested):
added some comments

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (27 of 27 strings)

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/es/

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.4

brendapaniagua reviewed (approved):
Functional Review! LGTM!

kittiu committed:
[13.0][FIX] purchase_sale_inter_company, PO->SO error when has line_section

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.5

kittiu commented:
Hi, please cherry-pick commit from this PR too, it fix some bug. https://github.com/OCA/multi-company/pull/276

Thanks!

PierrickBrun commented:
@kittiu Done and rebased

kevinkhao reviewed (approved):
LGTM

StefanRijnhart commented:
@PierrickBrun you cherry-picked https://github.com/OCA/multi-company/pull/276 but not @enriquemartin's https://github.com/OCA/multi-company/commit/eba7f0848a6, but then when resolving the conflict you included @enriquemartin's change into @kittiu's commit so that it is now included in the wrong commit with incorrect attribution. Could you redo the cherry-picking picking both commits this time (or simply the whole format-patch + precommit and then cherry-pick your own migration commit)?

Commit:
[IMP] purchase_sale_inter_company: black, isort, prettier

PierrickBrun commented:
@StefanRijnhart Thank you, I did the format-patch + cherry-pick

StefanRijnhart reviewed (approved):
Perfect, thank you for the update!

StefanRijnhart commented:
/ocabot merge nobump

legalsylvain commented:
/ocabot migration purchase_sale_inter_company
